src/index.js

```diff
@@ -176,16 +176,23 @@ async function extractViews(schemaName, db) {
 }
 
 /**
+ * @param {string | undefined} schemaName Name of the schema or `undefined` to
+ *   extract types from all schemas.
  * @param {Knex} db
  * @returns {Promise<Type[]>}
  */
-async function extractTypes(db) {
+async function extractTypes(schemaName, db) {
   /** @type {Type[]} */
   const types = [];
-  const enumTypes = await db
-    .select(['oid', 'typname'])
-    .from('pg_type')
-    .where('typtype', 'e');
+  const enumsQuery = db
+    .select(['t.oid', 't.typname'])
+    .from('pg_type as t')
+	.join('pg_namespace as n', 'n.oid', 't.typnamespace')
+    .where('t.typtype', 'e');
+  if (schemaName) {
+    enumsQuery.andWhere('n.nspname', schemaName);
+  }
```

Comment:
I forgot that `extractTypes` is not exported from this module. If you want to keep it as a local function only, then the condition for `undefined` `schemaName` is useless. Please let me know what do you prefer, I will update it.

Reply:
That's fine, we might change that in the future I guess :-)